src/main/java/com/boot/cart/service/CartService.java

```diff
@@ -185,7 +202,16 @@ public CartDTO removeProductFromCart(String email, String productName)
         cart.setTotal(cart.getTotal() - cart.getEntries().stream().filter(entry -> productName.equals(entry.getProductName())).findFirst().get().getQuantity() * productDTO.getPrice());
         cart.getEntries().removeAll(matchingEntries);
 
-        return cartEntityToDto(cartRepository.save(cart));
+        cartRepository.save(cart);
+
+        List<String> userCartEntries = cart.getEntries().stream().map(CartEntry::getProductName).collect(Collectors.toList());
+
+        String productParam = String.join(",", userCartEntries);
```

Comment:
you could extract this logic into a separate method(since you use it for 4 times). something to receive cartEntries and to return a list of product dtos.

Reply:
done! good point here aswell! 👍 👍